swarm/pss/prox_test.go

```diff
@@ -133,99 +123,132 @@ func readSnapshot(t *testing.T, nodeCount int) simulations.Snapshot {
 	return snap
 }
 
-func assingTestVariables(sim *simulation.Simulation, msgCount int) {
-	log.Debug("-------------------------------------------------------------------------")
-	for _, nodeId := range sim.NodeIDs() {
-		nodeAddrs[nodeId] = nodeIDToAddr(nodeId)
+func initializeTestData(d *testData, msgCount int) {
+	log.Debug("TestProxNetwork start")
+	d.nodeAddrs = make(map[enode.ID][]byte)
+	d.recipients = make(map[int][]enode.ID)
```

Comment:
Why not put init in `testData.new()` instead?

Reply:
yes make it in the constructor?

Reply:
```
func newTestData() *tstData {
    return &tstData{
        d.nodeAddrs: make(map[enode.ID][]byte),
        d.recipients: make(map[int][]enode.ID),
        d.allowed: make(map[int][]enode.ID),
        d.expectedMsgs: make(map[enode.ID][]uint64),
        d.allowedMsgs: make(map[enode.ID][]uint64),
        d.senders: make(map[int]enode.ID),
        handlerC: make(chan handlerNotification),
        doneC: make(chan struct{}),
        errC: make(chan error),
        msgC:  make(chan handlerNotification),
        kademlias: map[enode.ID]*network.Kademlia,
    }
}

func (d *tstData) init() {
    log.Debug("TestProxNetwork start")
    for _, nodeId := range d.sim.NodeIDs() {
            d.nodeAddrs[nodeId] = nodeIDToAddr(nodeId)
    }
    [....]
}

func testProxNetwork(t *testing.T) {
    testData := newTestData()
    [...]
}
```